### PR TITLE
Upgrade json-c 0.13.1

### PR DESCRIPTION
**Pull request by sodre**
TODO:
  - [x] windows builds correctly
  - [x] add windows test, i.e. test files are in the right place.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,23 +1,24 @@
...
   name: json-c
-  version: {{ version }}
+  version: {{ version.split('-')[0] }}
 
 source:
   fn: json-c-{{ version }}.tar.gz
-  url: https://s3.amazonaws.com/json-c_releases/releases/json-c-{{ version }}.tar.gz
-  sha256: 2a136451a7932d80b7d197b10441e26e39428d67b1443ec43bbba824705e1123
+  url: https://github.com/json-c/json-c/archive/json-c-{{ version }}.tar.gz
+  sha256:
```
Comment (ocefpaf) [inline recipe/meta.yaml]: Empty sha256

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,23 +1,24 @@
-{% set version = "0.12.1" %}
+{% set version = "0.13.1-20180305" %}
+{% set sha256 = "2a136451a7932d80b7d197b10441e26e39428d67b1443ec43bbba824705e1123" %}
```
Comment (ocefpaf) [inline recipe/meta.yaml]: No need for this Jinja variable, just add the Sha below.

**Review by ocefpaf — APPROVED** [review]

File: recipe/meta.yaml (hunk 2 above)
Comment (sodre) [reply recipe/meta.yaml]: will regro-cf-autotick-bot pick it up if its in the body of the meta.yaml?

Comment (ocefpaf) [reply recipe/meta.yaml]: Yep. And it makes the recipe clear and readable.

Comment (sodre) [reply recipe/meta.yaml]: okay I will revert :)

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,23 +1,24 @@
...
-    - automake
-    - libtool
+    - {{ compiler('c') }}
+    - autoconf  # [unix]
+    - automake  # [unix]
+    - libtool  # [unix]
+    - cmake  # [win]
 
 test:
   commands:
```
Comment (ocefpaf) [inline recipe/meta.yaml]: We need a Windows test. Also, can you take this opportunity and update the inspect tests to:

```yaml
    - conda inspect linkages -p $PREFIX $PKG_NAME  # [not win]
    - conda inspect objects -p $PREFIX $PKG_NAME  # [osx]
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,23 +1,30 @@
...
-    - automake
-    - libtool
+    - {{ compiler('c') }}
+  host:
+    - autoconf  # [unix]
+    - automake  # [unix]
+    - libtool  # [unix]
+
+    - cmake  # [win]
+    - msinttypes  # [win]
```
Comment (sodre) [inline recipe/meta.yaml]: @isuruf, 
   in the past I would have controlled the install of msinttypes with [win and py27]. Realistically, it is only needed for vc9. Do you know how to represent that constraint using conda-build-3?

Comment (isuruf) [reply recipe/meta.yaml]: `[win and vc<=9]`

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,23 +1,30 @@
...
-    - automake
-    - libtool
+    - {{ compiler('c') }}
+  host:
+    - autoconf  # [unix]
+    - automake  # [unix]
+    - libtool  # [unix]
+
+    - cmake  # [win]
+    - msinttypes  # [win and vc <=9]
```
Comment (isuruf) [inline recipe/meta.yaml]: All 5 packages above should be in build

File: recipe/meta.yaml [hunk 6]
```diff
@@ -1,23 +1,30 @@
...
+    - vc  # [win]
```
Comment (isuruf) [inline recipe/meta.yaml]: This is not needed

**Comment by sodre** [comment]
@isuruf, it looks like the number of builds for windows have doubled up now. Are you sure it is correct?

```
TESTS DURATION
Environment: CONFIG=win_c_compilervs2008vc14, CONDA_INSTALL_LOCN=C:\Miniconda36-x64
Queued
Environment: CONFIG=win_c_compilervs2008vc9, CONDA_INSTALL_LOCN=C:\Miniconda36-x64
Queued
Environment: CONFIG=win_c_compilervs2015vc14, CONDA_INSTALL_LOCN=C:\Miniconda36-x64
Queued
Environment: CONFIG=win_c_compilervs2015vc9, CONDA_INSTALL_LOCN=C:\Miniconda36-x64
Queued
```

**Comment by isuruf** [comment]
@conda-forge-admin, please rerender

**Comment by isuruf** [comment]
@conda-forge-admin, please rerender

**Comment by sodre** [comment]
adding the `- vc` line back works.

**Comment by sodre** [comment]
okay, awesome!

File: recipe/meta.yaml [hunk 7]
```diff
@@ -1,23 +1,28 @@
...
-    - autoconf
-    - automake
-    - libtool
+    - {{ compiler('c') }}
+    - autoconf  # [unix]
+    - automake  # [unix]
+    - libtool  # [unix]
+    - cmake  # [win]
+    - msinttypes  # [win and vc<=9]
+  host:
```
Comment (isuruf) [inline recipe/meta.yaml]: @msarahan, why is an empty `host` needed? To make sure `build` and `host` are separated?

**Pull request merged**